#! /usr/bin/env python

PACKAGE='yolo_ekf'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                Type       Lvl    Description                                   Default     Min     Max
gen.add("IoU_thresh",       double_t,  0,     "Maximum distance to associate measurement",  0.55,        0.1,    0.99)
gen.add("min_size_x",       double_t,  0,     "minimum bounding box width for tracking",    50.0,        0.1,    200.0)
gen.add("min_size_y",       double_t,  0,     "minimum bounding box height for tracking",   50.0,       0.1,    200.0)
gen.add("max_age",          double_t,  0,     "Max Age to delete ekf instance",             0.15,        0.0,    5.0)
gen.add("min_age",          double_t,  0,     "Max Age to delete ekf instance",             0.1,        0.0,    5.0)
gen.add("p_factor",         double_t,  0,    "Inital factor to set P diagonal to",          50000,        0.0,    100000.0)
gen.add("r_cx_cy",          double_t,  0,     "Kalman state standard deviation",            5.0,        1e-6,   100.0)
gen.add("r_w_h",            double_t,  0,     "Kalman state standard deviation",            10.0,        1e-6,   100.0)
gen.add("q",                double_t,  0,     "Kalman measurement standard deviation",      15.0,        1e-6,   100.0)

exit(gen.generate(PACKAGE, PACKAGE, "YOLOEkf"))
